packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -582,6 +554,27 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
     final List<Error> totalErrors = List<Error>.from(_errors);
     totalErrors.addAll(validateErrors);
 
+    for (final MapEntry<TypeDeclaration, List<int>> element
+        in referencedTypes.entries) {
+      if (!referencedClasses
+              .map((Class e) => e.name)
+              .contains(element.key.baseName) &&
+          !referencedEnums
+              .map((Enum e) => e.name)
+              .contains(element.key.baseName) &&
+          !validTypes.contains(element.key.baseName) &&
+          !element.key.isVoid &&
+          element.key.baseName != 'dynamic' &&
+          element.key.baseName.isNotEmpty) {
```

Comment:
Wouldn't it make more sense to put these three cheap checks first in the condition, to short-circuit the map/contains checks?

Reply:
The thought passed my mind, but I don't think they are also very likely.  So even if they are cheap if they aren't likely it's a waste to put them first.